apps/finance/contracts/Finance.sol

```diff
@@ -322,9 +345,14 @@ contract Finance is EtherTokenConstant, IsContract, AragonApp {
         recurringPaymentExists(_paymentId)
         transitionsPeriod
     {
-        require(_nextPaymentTime(_paymentId) <= getTimestamp64(), ERROR_EXECUTE_PAYMENT_TIME);
-
-        _executePayment(_paymentId);
+        uint256 paid = _executePayment(_paymentId);
+        if (paid == 0) {
+            if (_nextPaymentTime(_paymentId) > getTimestamp64()) {
+                revert(ERROR_EXECUTE_PAYMENT_TIME);
+            } else {
+                revert(ERROR_EXECUTE_PAYMENT_NUM);
+            }
```

Comment:
This could be written in 2 lines, as:
```
require(_nextPaymentTime(_paymentId) <= getTimestamp64(), ERROR_EXECUTE_PAYMENT_TIME);
revert(ERROR_EXECUTE_PAYMENT_NUM);
```
Not saying it is better this way, it just came up to my mind and found it kind of nice, but maybe it is clearer as it is. :wink:

Reply:
Another alternative is to use a ternary, but some quick checking with this on remix suggests the ternary is more expensive than the if/else 🤷‍♂️.